src/ol/interaction/modifyinteraction.js

```diff
@@ -976,31 +1067,36 @@ ol.interaction.Modify.prototype.insertVertex_ = function(segmentData, vertex) {
   }
 
   this.setGeometryCoordinates_(geometry, coordinates);
-  var rTree = this.rBush_;
   goog.asserts.assert(segment !== undefined, 'segment should be defined');
-  rTree.remove(segmentData);
   goog.asserts.assert(index !== undefined, 'index should be defined');
-  this.updateSegmentIndices_(geometry, index, depth, 1);
   var newSegmentData = /** @type {ol.ModifySegmentDataType} */ ({
     segment: [segment[0], vertex],
     feature: feature,
     geometry: geometry,
     depth: depth,
-    index: index
+    index: index,
+    origSegment: segmentData
   });
-  rTree.insert(ol.extent.boundingExtent(newSegmentData.segment),
-      newSegmentData);
   this.dragSegments_.push([newSegmentData, 1]);
 
   var newSegmentData2 = /** @type {ol.ModifySegmentDataType} */ ({
     segment: [vertex, segment[1]],
     feature: feature,
     geometry: geometry,
     depth: depth,
-    index: index + 1
+    index: index + 1,
+    isNew: true,
```

Comment:
Oh, that's actually not needed anymore since origSegment identifies this as a temporary segment created during dragging of a side.